spec/presenters/sort_presenter_spec.rb

```diff
@@ -0,0 +1,188 @@
+  subject(:presenter_with_sort) { described_class.new(content_item(sort_options: sort_options_without_relevance), {}) }
+  subject(:presenter_with_default) { described_class.new(content_item(sort_options: sort_options_with_default), {}) }
+
+  let(:no_sort_options) { nil }
+
+  let(:sort_options_without_relevance) {
+    [
+      { "name" => "Most viewed" },
+      { "name" => "Updated (newest)" }
+    ]
+  }
+
+  let(:sort_options_with_relevance) {
+    [
+      { "name" => "Most viewed" },
+      { "name" => "Updated (newest)" },
+      { "name" => "Relevance", "key" => "relevance" }
+    ]
+  }
+
+  let(:sort_options_with_default) {
+    [
+      { "name" => "Most viewed" },
+      { "name" => "Updated (oldest)", "default" => true }
+    ]
+  }
+
+  let(:sort_options_with_public_timestamp_default) {
+    [
+      { "name" => "Most viewed" },
+      { "name" => "Updated (newest)", "key" => "-public_timestamp", "default" => true }
+    ]
+  }
+
+  describe "#for_select" do
+    def sort_option(label, value, disabled: false, selected: false)
+      disabled_attr = disabled ? 'disabled="disabled" ' : ''
+      selected_attr = selected ? 'selected="selected" ' : ''
+      "<option data-track-category=\"dropDownClicked\" data-track-action=\"clicked\" data-track-label=\"#{label}\" #{disabled_attr}#{selected_attr}value=\"#{value}\">#{label}</option>"
+    end
+
+    it "returns an empty string when sort is not present" do
+      expect(presenter_without_sort.for_select).to eql("")
+    end
+
+    it "returns sort options without relevance when keywords is not present" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (newest)', 'updated-newest')
+      ].join("\n")
+
+      expect(presenter_with_sort.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with relevance disabled when keywords is blank" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (newest)', 'updated-newest'),
+        sort_option('Relevance', 'relevance', disabled: true)
+      ].join("\n")
+
+      presenter = described_class.new(content_item(sort_options: sort_options_with_relevance), {})
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with relevance enabled when keywords is not blank" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (newest)', 'updated-newest'),
+        sort_option('Relevance', 'relevance', disabled: false)
+      ].join("\n")
+
+      presenter = described_class.new(content_item(sort_options: sort_options_with_relevance), "keywords" => "something not blank")
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with no option selected when order is specified but does not exist in options" do
```

Comment:
Perhaps we should return sort options with the default specified, as in the test below, if order is specified but does not exist?

Reply:
Perhaps, yes. This behaviour is consistent with the existing behaviour, though so I think this change could be done in a separate commit.

https://github.com/alphagov/finder-frontend/pull/1069/files#diff-7c61a92169a590942f2d8b5226c6937dR120.